buildenv

```diff
@@ -33,18 +43,23 @@ zopen_append_to_env()
 cat <<ZZ
 if [ ! -z "\$ZOPEN_IN_ZOPEN_BUILD" ]; then
   export ZOPEN_EXTRA_CPPFLAGS="\${ZOPEN_EXTRA_CPPFLAGS} -DZOSLIB_OVERRIDE_CLIB=1"
-  export ZOPEN_EXTRA_CFLAGS="\${ZOPEN_EXTRA_CFLAGS} -fgnu-keywords -I\$PWD/include"
-  export ZOPEN_EXTRA_CXXFLAGS="\${ZOPEN_EXTRA_CXXFLAGS} -fgnu-keywords -I\$PWD/include"
+  export ZOPEN_EXTRA_CFLAGS="\${ZOPEN_EXTRA_CFLAGS} -isystem \$PWD/include"
+  export ZOPEN_EXTRA_CXXFLAGS="\${ZOPEN_EXTRA_CXXFLAGS} -isystem \$PWD/include"
   export ZOPEN_EXTRA_LDFLAGS="\${ZOPEN_EXTRA_LDFLAGS} -L\$PWD/lib"
-  export ZOPEN_EXTRA_LIBS="\${ZOPEN_EXTRA_LIBS} -lzoslib \$PWD/lib/CXXRT64.x \$PWD/lib/celquopt.s.o"
+  export ZOPEN_EXTRA_LIBS="\${ZOPEN_EXTRA_LIBS} -lzoslib"
+  if [[ "$ZOPEN_COMP" == "XL" ]]; then
```

Comment:
```suggestion
  if [[ "\$ZOPEN_COMP" == "XL" ]]; then
```

Reply:
We will also need \$PWD/lib/celquopt.s.o because it will get discarded when in an archive since there's no reference to it - https://ibm-systems-z.slack.com/archives/C493603M1/p1680282717818339

Reply:
I think that should really be `"$ZOPEN_COMP"`.  It is during the build process that we decide if CXXRT64.x should be added or not.  I'll move the if statement outside of the cat command and append to the end of the file.  

I'll add celquopt.s.o back in.  It got lost while merging.

Reply:
The git issue is that it is linking with a C link command but the zoslib archive is bringing in c++ code.  The link step needs to be a C++ link.  Use the `--driver-mode=g++` option in ZOPEN_EXTRA_LDFLAGS to force the link step to be c++.

Along that line, we should be able to replace the CXXRT64.x with `-F:xlclang++` for a cleaner set up.